Pages/registerPage.js

```diff
@@ -0,0 +1,227 @@
+    }
+
+    get emailField(){
+        return browser.driver.findElement(by.id("email"));
+    }
+
+    get passwordField(){
+        return browser.driver.findElement(by.id("password"));
+    }
+
+    get confirmPasswordField(){
+        return browser.driver.findElement(by.id("confirmPassword"));
+    }
+
+    get registerButton(){
+        return browser.driver.findElement(by.className("btn-submit"));
+    }
+
+    get registrationMessage(){
+        return browser.driver.findElement(by.css("div.status-content div"))
+    }
+
+    get hereLinkForLogin(){
+        return browser.driver.findElement(by.css("div.status-content div button"));
+    }
+
+    get firstNameValidationMessage(){
+        return browser.driver.findElement(by.css("div.form div:nth-child(2) small label"));
+    }
+
+    get lastNameValidationMessage(){
+        return browser.driver.findElement(by.css("div.form div:nth-child(3) small label"));
+    }
+
+    get emailValidationMessage(){
+        return browser.driver.findElement(by.css("div.form div:nth-child(4) small label"));
+    }
+
+    get passwordValidationMessage(){
+        return browser.driver.findElement(by.css("div.form div:nth-child(5) small label"));
+    }
+
+    get confirmPasswordMessage(){
+        return browser.driver.findElement(by.css("div.form div:nth-child(6) small label"));
+    }
+
+    get wordLogin(){
+        return browser.driver.findElement(by.className("login-word"));
+    }
+
+    get wordLoginLocator(){
+        return ".login-word"
+    }
+    
+    get statusErrorLocator(){
+        return ".status-error";
+    }
+
+    // ACTIONS
+
+    makeRandomEmail(){
+        console.log("This method makes random email");
+        return  `${data.email1}${Math.round(Math.random()*100)}6${Math.round(Math.random()*100)}${data.emailDomain}`;
+    }
+    makeRandomEmailWithoutdomain(dataEmail){
+        console.log("This method makes random email without domain");
+        return  `${data.email1}${Math.round(Math.random()*100)}${dataEmail}`;
+    }
+
+    waitForLoginWord(){
+        console.log("This method wait for word Login");
+        return browser.wait(EC.presenceOf($(this.wordLoginLocator)), 7000);
+    }
+
+    waitForStatusError(){
+        console.log("This method wait for status error");
+        return browser.wait(EC.presenceOf($(this.statusErrorLocator)), 7000);
+    }
+
+    clickOnWordLogin(){
+        console.log("This method clicks on word Login");
+        this.wordLogin.click();
+    }
+
+    clickOnHereLink(){
+        console.log("This method clicks on 'Here link'");
+        this.hereLinkForLogin.click();
+    }
+
+    getConfirmPasswordMessage(){
+        console.log("This method gets Confirm Password message");
+        return this.confirmPasswordMessage.getText();
+    }
+
+    getPasswordValidationMessage(){
+        console.log("This method validates Password message");
+        return this.passwordValidationMessage.getText();
+    }
+
+    getEmailValidationMessage(){
```

Comment:
Can all methods withy type "getXValidationMessage" be combined into one method "getValidationMessage(option)" which will have switch case logic implemented inside ?

Reply:
I tried to combine methods, where that combination is possible